cypress-io/cypress
# misc: Add missing properties to Cypress.spec type

- Closes https://github.com/cypress-io/cypress/issues/27835

### Additional details
- Why was this change necessary? _Exposes missing Cypress.spec type information in your IDE_
- What is affected by this change? _This should have no effect on the application, purely a type change_
- Any implementation details to explain? _Simply added the missing fields to where the Spec type is defined_

### Steps to test
I'm not able to compile the entire project locally at the moment, but in a sample project, changing the type information was enough to see these fields now can be auto-completed in an IDE.

### How has the user experience changed?
In IDE, additional type information should be displayed:
![2023-10-08T17:47:32,840821522+07:00](https://github.com/cypress-io/cypress/assets/660911/8f40c244-74e7-4dbc-8850-b1cf2742d156)


### PR Tasks


- [na] Have tests been added/updated?
- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)?
